services-request-input.md

```diff
@@ -66,6 +66,34 @@ If you would like to set a cookie before a response has been created, use the `C
 
     $cookie = Cookie::forever('name', 'value');
 
+#### Handling cookies without encryption
+
+If you don't want some cookies to be encrypted or decrypted, you can specify them in configuration.
+This is useful, for example, when you want to pass data from frontend to server side backend via cookies, and vice versa.
+
+Add names of the cookies that should not be encrypted or decrypted to `unencryptedCookies` parameter in the `config/cookie.php` configuration file.
+
+    /*
```

Comment:
Please update this comment to reflect the final one that was merged in.

Reply:
@pikanji ping

Reply:
@LukeTowers thanks, I just updated the comment.

Reply:
I need quick guideline.I have added following code in  under path    myplugin/config/config.php
but it is not working at all my_cookie is encrypting every time.
Please help

<?php

return [
   
  'cookie.unencryptedCookies' => [
      'my_cookie',
  ],

 'unencryptedCookies' => [
        'my_cookie',
    ],


];

Reply:
@thewebdevelopment It cannot be declared in `myplugin/config/config.php`, because configuration keys defined under plugin directories are prefixed with the plugin name.
It should be in `config/cookie.php` of your OctoberCMS project.
This file already exists in the latest OctoberCMS code. So, you can just add the key there.
https://github.com/octobercms/october/blob/master/config/cookie.php
If your project doesn't have it, you can just add this file as `config/cookie.php` and add your key there.

Or, if you want insert it from your plugin, you can inject it from your Plugin.php like below.

    public function boot()
    {
        Config::push('cookie.unencryptedCookies', "my_cookie");
    }

Also, make sure that your project has the october/rain library with version v1.0.438 or up.

Reply:
@pikanji 
Thanks for your reply.
I think I am missing october/rain library  in my system.But not sure why octobercms documenation did not include rain library.Any idea how we can install such library?

Reply:
@thewebdevelopment if you are missing the october/rain library then your october install would not run. If you are not using composer for your project, then it will just exist inside of your project's vendor/october/rain/src directory.

Reply:
@LukeTowers  Thanks for your reply.
I checked the vender folder for october/rain library and it was their.But cookie folder was missing.
So I download and uploaded the cookie folder at path vender/october\rain\src\Cookie
Already created config/cookie.php with required cookie information
Should I need to run any command to make it work?
I don't want to update all packages I just want to make cookie folder work only.
Please guide/

Reply:
@thewebdevelopment That's a terrible way to do that, I can't stress enough how likely that is to break at some point. If you really want to go down that route though just make sure you have all the changes from https://github.com/octobercms/library/pull/335

Reply:
@thewebdevelopment To answer you question about command, you don't have to run any command regarding the matter of cookie configuration, if you have the packages supporting `unencryptedCookies`.
About your modification to october/rain library, I agree with LukeTowers. I can't tell if it would work. From my experiences, modifying third party packages is the last resort. It will make your project very hard to maintain. I think updating packages would eventually be easier than maintaining code with modified packages, in the long run. I'm sorry, I don't think I can give you the solution you are expecting.

Reply:
Basically I just want to make it work( cookie/middleware ) and that is all I want.
if you guys have any possible way then please guide.

Reply:
@thewebdevelopment That error is telling you that it could not install the package. I've just added stable version to it. So, you can try the same command again to install it.

Reply:
@pikanji  Thanks for adding up **stable version** I appreciate that.